### PR TITLE
parallelize segmentation by chromosome.

### PR DESCRIPTION
**Pull request by brentp**
This adds parallelization by chromosome to segment.

I wasn't sure about the save_dataframe with processes>1, for that, can I just do, e.g.
```
s = io.StringIO()
data.to_csv(s)
```

?

### CODE REVIEW

File: cnvlib/segmentation/__init__.py [hunk 1]
```diff
@@ -12,11 +12,42 @@
...
+                                skip_low, skip_outliers,
+                                save_dataframe, rlibpath)
+
+    chroms = cnarr.chromosome.unique()
+    rets = []
+    # TODO: handle save_dataframe=True
+    with futures.ProcessPoolExecutor(processes) as pool:
+        for ret in pool.map(_ds, ((cnarr[cnarr.chromosome == c], method,
+                                   threshold, variants, skip_low, skip_outliers,
+                                   False, rlibpath) for c in chroms)):
```
Comment (etal) [inline cnvlib/segmentation/__init__.py]: You can do:

    ((chromarr, method, ...) for _c, chromarr in cnarr.by_chromosomes())

and avoid creating `chroms` above. (Not that this is a bottleneck, but it's cleaner.)

File: cnvlib/segmentation/__init__.py [hunk 2]
```diff
@@ -12,11 +12,42 @@
...
+            rets.append(ret)
```
Comment (etal) [inline cnvlib/segmentation/__init__.py]: Does this work differently than e.g. `rets = list(pool.map(...))`?

**Comment by etal** [comment]
For `save_dataframe`: That option is to print out the CBS dataframe as it would look in R, with minimal contact in Python -- mainly for debugging and exploration. So if you collect the `seg_out` strings from each chunk and then concatenate them however you can, that's ideal.

**Comment by brentp** [comment]
thanks for the review and comments. Latest commit addresses all AFAICT.

**Comment by etal** [comment]
Looks great, thanks so much!

**Pull request merged**